File: app/src/main/java/app/music/base/BaseMVVMPPresenter.java
package app.music.base;

import androidx.lifecycle.ViewModel;

public class BaseMVVMPPresenter<T extends ViewModel> {

    protected T mViewModel = null;

    public void attachViewModel(T viewModel) {
        mViewModel = viewModel;
    }

    public void detachViewModel() {
        mViewModel = null;
    }
}

File: app/src/main/java/app/music/base/BaseMVVMPPresenter.kt
package app.music.base

import androidx.lifecycle.ViewModel

abstract class BaseMVVMPPresenter<VM : ViewModel> {

    protected var mViewModel: VM? = null

    fun attachViewModel(viewModel: VM) {
        mViewModel = viewModel
    }

    fun detachViewModel() {
        mViewModel = null
    }
}
